jumpscale/sals/process/__init__.py

```diff
@@ -84,595 +141,833 @@ def execute(
 
-    Keyword Arguments:
-        sig {bool} -- os signal to send to the process(s) (default: {signal.SIGTERM.value})
-        match_predicate {callable} -- function that does matching between
-            found processes and the targested process, the function should accept
-            two arguments and return a boolean (default: {None})
+    Returns:
+        None or list of int: represents the IDs of the processes remaning alive.
     """
+    pids = get_pids(process_name, match_predicate=match_predicate)
+    failed_processes = []
+    for pid in pids:
+        try:
+            kill(pid, sig, timeout=timeout, sure_kill=sure_kill)
+        except (j.exceptions.Runtime, j.exceptions.Permission):
+            failed_processes.append(pid)
+    return failed_processes or None  # return None if the failed list is empty
+
+
+def kill_all_pids(pids, sig=signal.SIGTERM, timeout=5, sure_kill=False):
+    """Kill all processes with given pids.
 
-    pids = get_pids(name, match_predicate=match_predicate)
+    Args:
+        pids (list of int): The target processes IDs.
+        sig (signal, optional): See signal module constants. Defaults to signal.SIGKILL.
+        timeout (int, optional): How long to wait for a process to terminate (seconds) before raise exception\
+            or, if sure_kill=True, send a SIGKILL. Defaults to 5.
+        sure_kill (bool, optional): Whether to fallback to SIGKILL if the timeout exceeded for the terminate operation or not. Defaults to False.
+
+
+    Returns:
+        None or list of int: represents the IDs of the processes remaning alive.
+    """
+    failed_processes = []
     for pid in pids:
-        kill(pid, sig)
+        try:
+            kill(pid, sig, timeout=timeout, sure_kill=sure_kill)
+        except (j.exceptions.Runtime, j.exceptions.Permission):
+            failed_processes.append(pid)
+    return failed_processes or None  # return None if the failed list is empty
 
 
-def kill_process_by_port(port):
-    """Kill process by port
+def kill_process_by_port(port, ipv6=False, udp=False, sig=signal.SIGTERM, timeout=5, sure_kill=False):
+    """Kill process by port.
 
-    Arguments:
-        port (int) -- port number
+    Args:
+        port (int): The port number.
+        ipv6 (bool, optional): Whether to search the connections that using ipv6 instead of ipv4. Defaults to False.
+        udp (bool, optional): Whether to search the connections for UDP port instead of TCP. Defaults to False.
+        sig (signal, optional): See signal module constants. Defaults to signal.SIGTERM.
+        timeout (int, optional): How long to wait for a process to terminate (seconds) before raise exception\
+            or, if sure_kill=True, send a SIGKILL. Defaults to 5.
+        sure_kill (bool, optional): Whether to fallback to SIGKILL if the timeout exceeded for the terminate operation or not. Defaults to False.
+
+    Returns:
```

Comment:
No need to specify for `Returns` in docstring if it only returns `None`

Reply:
done